### PR TITLE
Feature/gateway

### PR DESCRIPTION
**Pull request by enclave-marc-barry**

### CODE REVIEW

File: docs/management/gateways.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+Gateway Access Policies obey trust requirements and ACLs the same was as a regular Direct Access Policies, so you can easilly apply Zero Trust requirements to the `Senders` connecting into your subnets, like multi-factor authentication.
+
+## Testing
+
+Check that you can ping your new Enclave Gateway from your Sender policy machine, use the Enclave Gateway's Enclave Virtual address (i.e. 100.64.69.156 in the example below). You can find the Enclave Virtual address of your Enclave Gateway by running `enclave status` on either system once connected by policy.
+
+You should also notice in the output of the `enclave status` CLI command on the sender-side systems that any routes advertised via your Enclave Gateway are listed under the appropriate peer, as shown below.
+
+If you can ping the Enclave Gateway (i.e. ping 100.64.0.70 in this example) via Enclave and your local system is showing 172.26.0.0/20 (or your equivalent subnet) as the Enclave Gateway for value against the correct peer, you should now be able to start sending traffic directly into that subnet via your new Enclave Gateway.
```
Comment (enclave-charlie) [inline docs/management/gateways.md]: "for value against the correct peer" - not sure what this means

File: docs/management/gateways.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+   Last activity . . . : 0.75 seconds ago
+   Transfer. . . . . . : 3.462 KB received, 4.071 KB sent, link rtt 0.59 ms
+   Virtual network . . : 100.64.0.0/10 (255.192.0.0)
+   Virtual address . . : 100.64.0.70
+   Gateway for . . . . : 172.26.0.0/20
+   Dns . . . . . . . . : wzl94.id.enclave, ubuntu-dev.enclave
+   ACLs. . . . . . . . : allow [icmp] from peer -> local, allow [any] from local -> peer
+```
+
+If you can ping the Enclave Gateway (i.e. `ping 100.64.0.70` in this example) via Enclave and your local system is showing 172.26.0.0/20 (or your equivalent subnet) as the Enclave Gateway for value against the correct peer, you should now be able to start sending traffic directly into that subnet via your new Enclave Gateway.
```
Comment (enclave-charlie) [inline docs/management/gateways.md:149]: "for value against the correct peer" - not sure what this means

File: docs/management/gateways.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+Knowing that the Enclave Gateway itself is reachable, try sending some traffic past the Enclave Gateway to a device in the subnet behind it. If the any of the host-based firewalls on the target systems not running Enclave allow it, you may be able to send pings as test of end-to-end connectivity.
```
Comment (enclave-charlie) [inline docs/management/gateways.md:151]: "as a test of end-to-end connectivity" or "as proof of end-to-end connectivity" maybe?

File: docs/management/gateways.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+In the example below we've successfully sent a ping from a Windows laptop in a coffee shop, via the Enclave Gateway at `100.64.0.70` in the office out to a printer with an IP address of `172.26.0.250` in the Gateway's local subnet.
```
Comment (enclave-charlie) [inline docs/management/gateways.md:153]: Bit hard to parse. Could we break up the sentence more? e.g

"In the example below, we've successfully sent a ping from a Windows laptop in a coffee shop, via the Enclave Gateway at `100.64.0.70` in the office, out to a printer with an IP address of `172.26.0.250` in the Gateway's local subnet."

File: docs/management/gateways.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+## Troubleshooting
+
+If you find an Enclave Gateway isn't working as expected, here's a simple troubleshooting checklist:
+
+1. Check your systems (sender systems and gateways) are all enrolled, connected and approved in the portal.
+
+2. Check that the sender systems(s) can ping the Gateway using the Gateway's Enclave address.
+
+3. Check the subnets which the Gateway is advertising
```
Comment (enclave-charlie) [inline docs/management/gateways.md:178]: Missing a full stop at the end.

File: docs/management/gateways.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+4. Check the output of `enclave status` on all systems has the correct `Gateway for` subnets.
+
+5. Check that the Gateway itself can reach (i.e. ping) other devices on its local subnet.
+
+6. Check the routing table has been correctly configured on the relevant Sender systems of the policy. 
+
+    The routing table is configured automatically by Enclave so unlikely to be the source of a problem unless there are other conflicting routes already in place. The `Interface` address is the client's local Enclave IP address.
```
Comment (enclave-charlie) [inline docs/management/gateways.md:186]: "so it's unlikely to be the source of a problem"

File: docs/management/gateways.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+    ```console
+    C:\> route print | findstr 172.26.0.0
+    IPv4 Route Table
+    ===========================================================================
+    Active Routes:
+    Network Destination         Netmask     Gateway         Interface    Metric
+    172.26.0.0            255.255.240.0     On-link    100.119.20.243        26
+    ```
+
+7. Check that iptables on the Enclave Gateway system is correctly configured
```
Comment (enclave-charlie) [inline docs/management/gateways.md:197]: is iptables singular? Reads a bit weirdly (feels like it should be "are correctly configured")

File: docs/management/gateways.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+    Run `sudo iptables -t nat -L -n -v`. 
+
+    In particular pay attention to the `to:` field on the postrouting chain, that should be the local (non-enclave) IP address of your Enclave Gateway.
```
Comment (enclave-charlie) [inline docs/management/gateways.md:201]: Think it should be "which" instead of "that"

File: docs/management/gateways.md (hunk 7 above)
Comment (enclave-charlie) [reply docs/management/gateways.md:197]: Also missing a full stop

**Review by enclave-charlie — CHANGES_REQUESTED** [review]
Ready pretty clearly to me (I didn't know what default routes were going in). Most of my comments are suggestions, but also found a few typos.

File: docs/management/gateways.md [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+
+However there may be times when you do want Enclave to route your public Internet traffic, like ensuring a predictable and static IP address is used to access to trusted SaaS services such as Office 365 or Salesforce.
+
+To setup an Enclave Gateway as a default route on your network you should:
+
+1.	Enrol a Linux system to act as your Enclave Gateway
+
+2.	In the Portal, allow that enrolled system to act as an Gateway
+
+3.	Manually add the subnet `0.0.0.0/0` to that system in the Portal
```
[the remark this answers is not in the document]
Comment (enclave-marc-barry) [reply docs/management/gateways.md:108]: Not sure how to fix, want to plant the use-case seed

File: docs/management/gateways.md (hunk 1 above)
Comment (enclave-marc-barry) [reply docs/management/gateways.md]: Haha, nor me :p
